sematic/resolvers/local_resolver.py

```diff
@@ -182,14 +204,28 @@ def _notify_pipeline_update(self):
         root_future = self._futures[0]
         api_client.notify_pipeline_update(self._runs[root_future.id].calculator_path)
 
-    def _resolution_did_succeed(self) -> None:
-        super()._resolution_did_succeed()
+    def _resolution_did_succeed(self, root_future: AbstractFuture) -> None:
+        super()._resolution_did_succeed(root_future)
+        self._update_resolution_status(root_future.id, ResolutionStatus.COMPLETE)
         self._notify_pipeline_update()
 
-    def _resolution_did_fail(self) -> None:
-        super()._resolution_did_fail()
+    def _resolution_did_fail(
+        self, root_future: AbstractFuture, due_to_calculator_error: bool
+    ) -> None:
+        super()._resolution_did_fail(root_future, due_to_calculator_error)
+        resolution_status = (
+            ResolutionStatus.COMPLETE
+            if due_to_calculator_error
+            else ResolutionStatus.FAILED
+        )
+        self._update_resolution_status(root_future.id, resolution_status)
         self._notify_pipeline_update()
 
+    def _update_resolution_status(self, root_id: str, status: ResolutionStatus):
```

Comment:
`root_id` could be gotten from `self._root_future` if we had one.

Alternatively, we could persist `resolution` as `self._resolution` as soon as we create it. Then we wouldn't need to re-fetch it every time.

Reply:
> root_id could be gotten from self._root_future if we had one.

done

> Alternatively, we could persist resolution as self._resolution as soon as we create it. Then we wouldn't need to re-fetch it every time.

That might work, but I tend to lean towards keeping "source of truth" being what's in the DB, and referring to that here. That might become relevant when we start trying to handle cases where the resolver can be interrupted and restarted.